### PR TITLE
Fixes Stamcritting Not Making The User Incapacitated

### LINKED ISSUE #50325: You can engage in unarmed combat even when in stamcrit
Title: you can be in stamcrit and still be able to shove people and punch them. Makes arresting nigh on impossible and necessitates using lethal force, pretty bad.

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
:cl:Cobby
fix: You are now truly incapacitated while stamcritted.
/:cl:

~~Disarm checks for knockdown, but stamina does paralyze.~~

Fixes #50325

### CODE REVIEW

**Comment by Tumesuo** [comment]
what about being able to punch while in stamcrit too? will this fixie?

**Review by Rohesie — DISMISSED** [review]
I really dislike this non-generic behavior, but a fix is a fix.
I'll get around refactoring it in time.

**Comment by Akrilla** [comment]
Has this always been the case then?

**Comment by ExcessiveUseOfCobblestone** [comment]
this is not the right fix

**Comment by ExcessiveUseOfCobblestone** [comment]
Basically since stam_paralyze is separate from true Paralysis, I've added (and removed) the trait where one begins and ends stamcrit @Rohesie

**Comment by ExcessiveUseOfCobblestone** [comment]
> 
> 
> what about being able to punch while in stamcrit too? will this fixie?

This will fix most issues regarding stam not acting like paralysis, so "Yes and more"

**Review by Rohesie — DISMISSED** [review]
Much better.

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -521,8 +521,10 @@
 	var/stam = getStaminaLoss()
 	if(stam > DAMAGE_PRECISION && (maxHealth - stam) <= crit_threshold && !stat)
 		enter_stamcrit()
+		ADD_TRAIT(src, TRAIT_INCAPACITATED, STAMINA)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon.dm:524]: Maybe put the `ADD_TRAIT` where the `stam_paralyzed` variable changes?

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/carbon/carbon.dm:524]: you're right since `enter_stamcrit` can early return.

**Review by Rohesie — APPROVED** [review]
Third time's the charm.

**Pull request merged**